lib/index.js

```diff
@@ -55,82 +55,79 @@ function setupRateLimitEventExt(server, options) {
   const extractKey = options.extractKey;
   const onError = options.onError;
 
-  const extractKeyAndTakeToken = function(limitd, request, reply, type) {
-    extractKey(request, reply, (err, key) =>{
-      if (err) { return reply(err); }
-
-      if (!limitd) {
-        // limitd is not connected, do not fail!
-        return reply.continue();
-      }
-
-      limitd.take(type, key, (err, currentLimitResponse) => {
-        if (err){
-          if (onError) { return onError(err, reply); }
-          // by default we don't fail if limitd is unavailable
-          return reply.continue();
-        }
-
-        const oldMinimumLimitResponse = request.plugins.patova && request.plugins.patova.limit
-        const newMinimumLimitResponse = getMinimumLimit(currentLimitResponse, oldMinimumLimitResponse)
-
-        request.plugins.patova = request.plugins.patova || {};
-        request.plugins.patova.limit = newMinimumLimitResponse;
-
-        if (newMinimumLimitResponse.conformant) {
-          // We continue only if the request is conformat so far
-          return reply.continue();
-        }
-
-        const error = Boom.tooManyRequests();
-        error.output.headers = new RateLimitHeaders(
-          newMinimumLimitResponse.limit,
-          newMinimumLimitResponse.remaining,
-          newMinimumLimitResponse.reset);
-
-        reply(error);
-      });
-    });
+  const extractKeyAndTakeToken = async function(limitd, request, handler, type) {
+    const key = extractKey(request, handler);
```

Comment:
Shouldn't `extractKey` return a promise? Since it was async before. Also, it seems that before this might not call the callback, preventing the rest of the method from executing: https://github.com/auth0/auth0-mfa-api/pull/707/files#diff-9c6bd636db8132f8f1ec1372df14a63bL33 . If extractKey can return `h.continue` we might need to check if `key === h.continue` before proceeding.